src/sentry/mail/adapter.py

```diff
@@ -280,27 +280,33 @@ def disabled_users_from_project(project: Project) -> Mapping[ExternalProviders,
 
     def get_send_to_team(self, project, target_identifier):
         if target_identifier is None:
-            return []
+            return {}
         try:
             team = Team.objects.get(id=int(target_identifier), projectteam__project=project)
         except Team.DoesNotExist:
-            return set()
+            return {}
 
-        disabled_users = self.disabled_users_from_project(project).get(ExternalProviders.EMAIL)
-        if disabled_users:
-            return set(team.member_set.values_list("user_id", flat=True)) - disabled_users
-        else:
-            return set(team.member_set.values_list("user_id", flat=True))
+        member_list = team.member_set.values_list("user_id", flat=True)
+        users = User.objects.filter(id__in=member_list)
+        team_members_by_provider = NotificationSetting.objects.filter_to_subscribed_users(
+            project, users
+        )
+        output = defaultdict(list)
```

Comment:
Because the inner loop is just adding every user_id, you can replace this block of code with 
```
return {
    provider: [user.id for user in team_members]
    for provider, team_members in team_members_by_provider.items()
}
```